src/MongoDB.Driver.Core/Core/Servers/Server.cs

```diff
@@ -372,9 +445,9 @@ private bool ShouldInvalidateServer(Exception exception)
                 var code = (ServerErrorCode)commandException.Code;
                 var message = commandException.ErrorMessage;
 
-                if (IsNotMasterOrRecovering(code, message))
+                if (IsStateChangeError(code, message))
```

Comment:
Rename of this function per the [spec commit](https://github.com/mongodb/specifications/commit/7dd9c008be6edaf65368787d5e91c2ac3aeb95b8#diff-62b4da2e38e7d2a9c1a2d92d315895bbR1357-R1358).